auth/services/oauth/holder.go

```diff
@@ -102,14 +106,20 @@ func (v *HolderService) ClientMetadata(ctx context.Context, endpoint string) (*o
 	return metadata, nil
 }
 
-func (v *HolderService) PostError(ctx context.Context, auth2Error oauth.OAuth2Error, verifierResponseURI string) (string, error) {
+func (v *HolderService) PostError(ctx context.Context, auth2Error oauth.OAuth2Error, verifierResponseURI string, verifierClientState string) (string, error) {
 	iamClient := iam.NewHTTPClient(v.strictMode, v.httpClientTimeout, v.httpClientTLS)
 
 	responseURL, err := core.ParsePublicURL(verifierResponseURI, v.strictMode)
 	if err != nil {
 		return "", fmt.Errorf("failed to post error to verifier: %w", err)
 	}
-	redirectURL, err := iamClient.PostError(ctx, auth2Error, *responseURL)
+	validURL := *responseURL
+	if verifierClientState != "" {
+		validURL = http.AddQueryParams(*responseURL, map[string]string{
+			oauth.StateParam: verifierClientState,
+		})
+	}
+	redirectURL, err := iamClient.PostError(ctx, auth2Error, validURL)
 	if err != nil {
```

Comment:
do we need to check the redirect URL, whether it's ours? (do we trust the AS)

Reply:
Can't find it in the security considerations of the spec(s). But won't hurt to add.